### PR TITLE
Google authentication landing page

### PR DESCRIPTION
**Pull request by nikhen-s**
## Objective
Implement Google Authentication for users to be able to view and modify required information to use the app

## Key Changes
- [x] Aligned the Login Page UI to look like the Figma Mockup
- [x] Button on the landing page successfully calls the Google Authentication API
- [x] Upon successful authentication, the app navigates to home page
- [x] Upon failing authentication, the app navigates back to login page (and shows an error message in a AlertDialog Box)

Related to Issue https://github.com/swent-sp-2024-event-radar/event-radar-app/issues/13

### CODE REVIEW

File: app/google-services.json
Comment (jgeaso1266) [inline app/google-services.json:15]: It looks like these changes are still in this PR, can you move this to your other PR?

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:66]: The standard way to do this is using the `by` keyword (then you won't need that `@SuppressLint` annotation.

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:117]: nit: it might be better to use constraints instead of just dp values to adapt to different screen sizes.

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:111]: nit: the color should be using the material design options like `primary` and `primaryContainer` as opposed to hardcoding.

File: app/src/main/res/layout/activity_firebase_uiactivity.xml
Comment (jgeaso1266) [inline app/src/main/res/layout/activity_firebase_uiactivity.xml:7]: nit: I don't believe this needs to be commented out (even though it was red). This is generated by firebase I believe.

File: app/src/main/res/values/strings.xml
Comment (jgeaso1266) [inline app/src/main/res/values/strings.xml:3]: This should move to the other PR as well

**Review by jgeaso1266 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:97]: You should modify the color to work with both light and dark mode (using the material design colors).

**Review by jgeaso1266 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (christinehahaa) [inline app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:1]: Nice!

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (christinehahaa) [inline app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:77]: Nice!

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (christinehahaa) [inline app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:58]: nitpick: I'm thinking "Ok" instead of "Confirm" makes more sense? We didn't have a Figma mockup for this, but it's a small thing so no need for one.

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (Viccha18) [reply app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:77]: Good additon to handle the failure case

**Review by Viccha18 — APPROVED** [review]
Good job ! Nothing to add from Jalen and Christine. Just a reminder for later : the new route to homescreen will be `Route.EVENTS` after we merged the new navigation bar PR

File: app/google-services.json
Comment (nikhen-s) [reply app/google-services.json]: Done, Files Changed now doesn't show this :)

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (nikhen-s) [reply app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:66]: DM'ed, and solution is ok it turns out 👍

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (nikhen-s) [reply app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:117]: Changed to constraints layout in latest commit.

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (nikhen-s) [reply app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:111]: Done as well, just waiting for Vicky's PR to be merged so that I can use primaryContainer colour.

File: app/src/main/res/layout/activity_firebase_uiactivity.xml
Comment (nikhen-s) [reply app/src/main/res/layout/activity_firebase_uiactivity.xml:7]: Done :)

File: app/src/main/res/values/strings.xml
Comment (nikhen-s) [reply app/src/main/res/values/strings.xml]: 👍

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (nikhen-s) [reply app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:97]: 👍

File: app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt
Comment (nikhen-s) [reply app/src/main/java/com/github/se/eventradar/ui/login/LoginScreen.kt:58]: 👍

**Comment by nikhen-s** [comment]
Newest Commits: All the changes are made, 
Vicky's highlight on the new route to homescreen will be Route.EVENTS, I've added as a comment next to Route.Overview since waiting for Navigation Bar PR to merge to main + Japen's highlight on using Material Theme Colours, I have changed everything with exception for the primaryContainer for button since waiting on Vicky's Colours PR to merge to main.

**Pull request merged**
